plugins/@grouparoo/app-templates/src/source/table/tableSpeculation.ts

```diff
@@ -0,0 +1,40 @@
+import { PropertyTypes } from "@grouparoo/core";
+
+export namespace TableSpeculation {
+  export function isUniqueColumn(columnName: string) {
+    columnName = columnName.toLowerCase();
+
+    if (
+      [
+        "email",
+        "email_address",
+        "id",
+        "user_id",
+        "userid",
+        "uuid",
+        "guid",
+      ].includes(columnName)
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  export function columnType(
+    columnName: string,
+    databaseType: typeof PropertyTypes[number]
+  ): typeof PropertyTypes[number] {
+    columnName = columnName.toLowerCase();
+
+    if (databaseType !== "string") {
+      return databaseType;
+    }
+
+    if (columnName.includes("email")) {
+      return "email";
+    }
+
+    return "string";
+  }
+}
```

Comment:
@bleonard I created this helper to try to guess some additional information about the column defaults.  I'm not really sure where this belongs...